programs/nft-club/src/components/creator.rs

```diff
@@ -0,0 +1,99 @@
+use anchor_lang::prelude::*;
+
+// 
+// Endpoints
+// 
+
+/*
+ * Create account and benefit in same function
+ * creator specifies the actual benefits in form when sign up
+ *
+ * Can program instructions handle variable amount of benefits
+ * Benefit seeds are passed in here as parameters
+ */
+/* program.rpc.createAccount on frontend*/
+pub fn create_account(ctx: Context<CreateAccount>, username: String, email: String, description: String, num_benefits: u8) -> ProgramResult {
+    let creator: &mut Account<Creator> = &mut ctx.accounts.creator;
+    let authority: &Signer = &ctx.accounts.authority;
+
+    if username.chars().count() > 42 {
+        return Err(ErrorCode::UsernameTooLong.into())
+    }
+
+    if email.chars().count() > 42 {
+        return Err(ErrorCode::EmailTooLong.into())
+    }
+
+    if description.chars().count() > 420 {
+        return Err(ErrorCode::DescriptionTooLong.into())
+    }
+
+    creator.authority = *authority.key;
+    creator.username = username;
+    creator.email = email;
+    creator.description = description;
+    creator.num_benefits = num_benefits;
```

Comment:
We probably shouldn't include this value in this function, because it would be weird to ask them how many benefits they want without asking what they are.

So we can just increment this value when we call `create_benefit`.

EDIT: nvm, it makes sense in the scenario where the Creator fills out a single form upon signup, because we'll infer the count at that time rather than ask them explicitly for the number 👍